contrib/debug-tools/docs/remote-debug-pai-job.md

```diff
@@ -0,0 +1,85 @@
+# 1. Overview
+
+OpenPai remote debug tool only support **python** job. It contains two debug modes:
+
+1. Start remote debug when job task starts
+2. Inject breakpoint in source code and start debug when hit the breakpoint.
+
+Currently this feature is in experiment phrase, and we don't have a mechanism to notify user weather debug server is starts. Users need to check log to my theirself to get this notification.
+
+We recommend to use first approach to start debug. To use second approach, user need to manage the debug ports by themselves.
+
+***NOTICE: This is an experiment feature and may be changed in future release***
+
+# 2. Edit the job submission yaml to support debug
+To support the live debug, we need to modify the job submit yaml. Please request one debug port in `resourcePerInstance` field.
+And create a new `debug` deployment.
+
+```yaml
+taskRoles:
+  train:
+    resourcePerInstance:
+      cpu: 2
+      memoryMB: 16384
+      gpu: 4
+      ports:
+        debug: 1
+deployments:
+  - name: debug
+    taskRoles:
+      train:
+        preCommands:
+          - >-
+            if [ ! -d /pai_data/debug ]; then mkdir --parents /pai_data/debug; fi
+          - apt-get install -y --no-install-recommends wget
+          - >-
+            wget
+            https://raw.githubusercontent.com/microsoft/pai/master/contrib/debug-tools/openpaipdb/paipdb.py
+            -P /pai_data/debug
+
+defaults:
+  deployment: debug
+``` 
+
+The full examples can refer to [examples](../examples).
+
+# 3. Start debugging when task begins
+To start debugging when task begins. We need to change the task role command.
+
+Here is a sample:
+```bash
+PYTHONPATH=$PYTHONPATH:/pai_data/debug DEBUG_PORT_NAME=your_debug_port_name DEBUG_TIMEOUT=600 python -m paipdb user_python_script.py args ...
+```
+or you can use python3 to run the script:
+```bash
+PYTHONPATH=$PYTHONPATH:/pai_data/debug DEBUG_PORT_NAME=your_debug_port_name DEBUG_TIMEOUT=600 python3 -m paipdb user_python_script.py args ...
+```
+
+The `DEBUG_PORT_NAME` here is the debug port we request in `resourcePerInstance`. The `DEBUG_TIMEOUT` means the seconds debug server will wait for connecting. If there is no client connect to debug server before timeout, the task will exit.
+
+# 4. Inject breakpoint to debug
+If user want to start debug at somewhere. He/she can inject the breakpoint into the code. When program hint such breakpoint, the debug server will start. Then user can connect the debug server and start remote debug.
+
+This is useful for remote debug multithread process.
+
+To enable this, user can set breakpoint by insert `import paipdb; paipdb.settrace(port=debug_port)`.
```

Comment:
set paipdb.settrace(port=debug_port) to where user wants to break? Better separate import and inject breakpoints

Reply:
Refer to [pdb doc](https://docs.python.org/3/library/pdb.html):
```
The typical usage to break into the debugger from a running program is to insert
import pdb; pdb.set_trace()
```

I think it's ok to  keep this here